[Peanuts359] iP

Peanuts359 opened the pull request:
> # That One Guy
> > “You don’t need a new plan for next year. You need a commitment.” – Seth Godin ([source](link-to-source))
> 
> A Java application that seems to absolutely hate its job.
> Nevertheless, it will still help keep track of tasks so you don't have to.
> 
>The application is:
> * text-based
> * easy to learn
> * ~FAST~ ~SUPER FAST~ __REALLY FAST__ to use
> 
> To use it, you need to;
> 
> 1. download it from [here](https://github.com/peanuts359/ip).
> 2. double-click it.
> 3. add your tasks.
> 4. let it manage your tasks for you
> 
> And despite its attitude, it charges **NOTHING**.
> 
> Features:
> 
> * [x]  Managing tasks
> * [x]  A working GUI
> * [ ]  Built-in documentation
> 
> If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
> 
> ```java
> public class Main {
>     public static void main(String[] args) {
>         Application.launch(MainApp.class, args);
>     }
> }
> ```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Peanuts359 committed:
Reworked greeting

Peanuts359 committed:
Revert "Reworked greeting"

This reverts commit a7d84c2bb1228ed8092658b2e387104edcf780fa.

Peanuts359 committed:
Reworked greeting

Peanuts359 committed:
Reworked greeting

Peanuts359 committed:
Added ability to echo commands sent in terminal by a user

Peanuts359 committed:
Replaced echo functionality with the ability to add and view tasks.
Also made the chatbot really rude for no apparent reason.

Peanuts359 committed:
Added an indicator for task completion

Peanuts359 committed:
Added ability to mark tasks as done / not done

Peanuts359 committed:
Added new Task types

Peanuts359 committed:
Properly implemented Task types

Peanuts359 committed:
Clean up code, add error handling

Peanuts359 committed:
Add automated testing

Peanuts359 committed:
Add deletion functionality

Peanuts359 committed:
Add support for reading from / writing to a file

Peanuts359 committed:
Merge branch 'branch-Level-7'

Peanuts359 committed:
Add proper date and time support

Peanuts359 committed:
Merge branch 'branch-Level-8'

Peanuts359 committed:
Reorganise code

Peanuts359 committed:
Merge branch 'branch-A-MoreOOP'

Peanuts359 committed:
Add more package types

Peanuts359 committed:
Merge branch 'branch-A-Packages'

Peanuts359 committed:
Merge branch 'add-gradle-support'

Peanuts359 committed:
Add gradle support for building/running

Peanuts359 committed:
Reorganise classes

Peanuts359 committed:
Add test support

Peanuts359 committed:
Merge branch 'branch-A-JUnit'

Peanuts359 committed:
Minor tweak

Peanuts359 committed:
Add documentation of every non-trivial method, via JavaDoc

Peanuts359 committed:
Merge branch 'branch-A-JavaDoc'

Peanuts359 committed:
Update code as per Java coding standards

Peanuts359 committed:
Merge branch 'branch-A-CodingStandard'

Peanuts359 committed:
Add support for searching for tasks using keywords

Peanuts359 committed:
Merge branch 'branch-Level-9'

Peanuts359 committed:
Cleanup code using checkstyle

Peanuts359 committed:
Merge branch 'branch-A-CheckStyle'

Peanuts359 committed:
Add Varargs support

Peanuts359 committed:
Merge branch 'branch-A-Varargs'

Peanuts359 committed:
Layout some assets, add enum support

amitha2210 commented on src/main/java/guy/tasks/TaskManager.java:
```diff
@@ -0,0 +1,262 @@
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
```
Maybe you can split the java.util.ArrayList import statement from the first 3 with an empty line?

Peanuts359 committed:
Modify code, to add UI content in next commit

amitha2210 commented on src/test/java/guy/parser/ParserTest.java:
```diff
@@ -0,0 +1,64 @@
+        Parser p = new Parser(new Scanner(new ByteArrayInputStream("nutmeg".getBytes())));
+        outContent.reset();
+        p.cmd();
+        assertEquals("Maybe put in an actual command next time, shitass.", outContent.toString().trim());
```
Maybe you can add a line break here?

amitha2210 commented on src/main/java/guy/parser/Parser.java:
```diff
@@ -0,0 +1,87 @@
+     *
+     * @return false if the user requests to quit, true otherwise
+     */
+    public boolean cmd() {
```
Maybe can rename cmd to fit the coding standard better since it's a boolean?

amitha2210 commented on src/main/java/guy/tasks/TaskManager.java:
```diff
@@ -0,0 +1,262 @@
+            }
+            Storage.saveData();
+            System.out.println(tasks.get(id).toString());
+        }
```
I like this piece of code

amitha2210 reviewed (commented):
Looks good 👍

Peanuts359 committed:
Fully implement the GUI

Peanuts359 committed:
Merge branch 'branch-Level-10'

MingYiAw commented on src/main/java/guy/ThatOneGuy.java:
```diff
@@ -0,0 +1,83 @@
+ * It initialises a list of tasks, the UI, and storage, and handles the main logic.
+ */
+public class ThatOneGuy {
+    private TaskManager tm;
```
Perhaps a more intuitive name for `TaskManager` here?

MingYiAw commented on src/main/java/guy/ThatOneGuy.java:
```diff
@@ -0,0 +1,83 @@
+        ui = new Ui();
+        storage = new Storage();
+        tm = new TaskManager();
+        this.parser = new Parser(new Scanner(System.in));
```
Perhaps `this` keyword could be remove here?

MingYiAw commented on src/main/java/guy/ThatOneGuy.java:
```diff
@@ -0,0 +1,83 @@
+    /**
+     * Tells the application to continue reading user inputs, until the user requests to quit.
+     */
+    private void keepGoing() {
```
I like how you name this method, interesting name.

MingYiAw commented on src/main/java/guy/exception/GuyException.java:
```diff
@@ -0,0 +1,25 @@
+    @Override
+    public String getMessage() {
+        return super.getMessage();
+    }
```
Is this overriding of `getMessage()` method really needed? 
By default, when a class extends `Exception`, it already inherits the `getMessage()` method from the `Exception` class, which is same as `super.getMessage()`

MingYiAw commented on src/main/java/guy/tasks/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+/**
+ * Representation of a task with a deadline.
+ */
+
```
Perhaps this empty line should be remove?

MingYiAw commented on src/main/java/guy/tasks/Task.java:
```diff
@@ -0,0 +1,73 @@
+ */
+public abstract class Task {
+    protected String name;
+    protected boolean isDone;
```
I like how you name booleans following the naming rules. i.e. start with 'is'

MingYiAw commented on src/main/java/guy/tasks/TaskManager.java:
```diff
@@ -0,0 +1,269 @@
+ */
+public class TaskManager {
+    private static TaskManager tm;
+    private final ArrayList<Task> tasks;
```
I like how you name multi-valued variables in plural form.

MingYiAw commented on src/main/java/guy/tasks/TaskManager.java:
```diff
@@ -0,0 +1,269 @@
+            case "D" -> new Deadline(line[2], LocalDateTime.parse(line[3], dtf));
+            case "E" -> new Event(line[2], LocalDateTime.parse(line[3], dtf), LocalDateTime.parse(line[4], dtf));
+            default -> throw new GuyException("Why did you give me a file with an invalid line, you dingus...");
+            };
```
Perhaps the `switch` statement should be in the form mention in coding standard?

MingYiAw reviewed (commented):
LGTM, Maybe some minor changes need to be made for some parts. I like how you write the messages for **ThatOneGuy**

Peanuts359 committed:
Add assert statements to ThatOneGuy.java and Parser.java.

This verifies that the preprocessing steps work correctly.

Peanuts359 committed:
Tidy up code in the various classes of ThatOneGuy

Notably, renamed some variables to be more self-explanatory.
Also removed unused variables.

Peanuts359 committed:
Merge pull request #1 from Peanuts359/branch-A-Assertions

Add assert statements to ThatOneGuy.java and Parser.java.

Peanuts359 committed:
Merge branch 'master' into branch-A-CodeQuality

Peanuts359 committed:
Merge pull request #2 from Peanuts359/branch-A-CodeQuality

Tidy up code in the various classes of ThatOneGuy

Peanuts359 committed:
Rewrite TaskManager.java to use Streams

Peanuts359 committed:
Merge pull request #3 from Peanuts359/branch-A-Streams

Rewrite TaskManager.java to use Streams

Peanuts359 committed:
Add support for the help command

Every command has a help page, including the help command itself.
The help page gives a description and the syntax of the command.
Failing to provide an input instead pulls up syntax for all commands.
Descriptions are a work in progress.

Peanuts359 committed:
Finish up Help.java

Peanuts359 committed:
Merge branch 'branch-BCD-Extension'

Peanuts359 committed:
Rework some tests

Peanuts359 committed:
Clean up more code

Peanuts359 committed:
Add a product screenshot

Peanuts359 committed:
Partially write user guide, to complete later

Peanuts359 committed:
Write out the remaining sections of the user guide

Peanuts359 committed:
Rewrite description

Peanuts359 committed:
Fix documentation

Peanuts359 committed:
Resolve conflict in README.md
